checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -200,7 +201,14 @@ def _build_key_to_val_changes(new_val: str, new_key: str):
             EACH_KEY: new_key
         }
 
-    def _create_new_resource(self, main_resource: TerraformBlock, new_value: int | str, new_key: Optional[str] = None):
+    def _create_new_resource(
+            self,
+            main_resource: TerraformBlock,
```

Comment:
you don't need to get the main_resource as a parameter, you are getting the `resource_idx` so you can get it directly from the graph